pkg/test/e2e/kubefed.go

```diff
@@ -0,0 +1,174 @@
+package e2e
+
+import (
+	"context"
+	"github.com/codeready-toolchain/api/pkg/apis"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"os"
+	"sigs.k8s.io/kubefed/pkg/apis/core/common"
+	"sigs.k8s.io/kubefed/pkg/apis/core/v1beta1"
+	"testing"
+	"time"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	OperatorRetryInterval = time.Second * 5
+	OperatorTimeout       = time.Second * 60
+	RetryInterval         = time.Millisecond * 100
+	Timeout               = time.Second * 3
+	CleanupRetryInterval  = time.Second * 1
+	CleanupTimeout        = time.Second * 5
+	MemberNsVar           = "MEMBER_NS"
+	HostNsVar             = "HOST_NS"
+)
+
+// VerifyKubeFedCluster verifies existence and correct conditions of KubeFedCluster CRD
+// in the target cluster type operator
+func VerifyKubeFedCluster(t *testing.T, targetClusterType cluster.Type) {
+	// given
+	fedClusterList := &v1beta1.KubeFedClusterList{}
+	ctx, awaitility := InitializeOperators(t, fedClusterList, targetClusterType)
+	defer ctx.Cleanup()
+
+	var kubeFedClusterType cluster.Type
+	var singleAwait *SingleAwaitility
+	var expNs string
+
+	if targetClusterType == cluster.Host {
+		kubeFedClusterType = cluster.Member
+		singleAwait = awaitility.Host()
+		expNs = awaitility.MemberNs
+	} else {
+		kubeFedClusterType = cluster.Host
+		singleAwait = awaitility.Member()
+		expNs = awaitility.HostNs
+	}
+
+	current, ok, err := singleAwait.GetKubeFedCluster(expNs, kubeFedClusterType, nil)
+	require.NoError(awaitility.T, err)
+	require.True(awaitility.T, ok, "KubeFedCluster should exist")
+	labels := KubeFedLabels(kubeFedClusterType, current.Labels["namespace"], current.Labels["ownerClusterName"])
+
+	t.Run("create new KubeFedCluster with correct data and expect to be ready", func(t *testing.T) {
+		// given
+		newName := "new-ready-" + string(kubeFedClusterType)
+		newFedCluster := NewKubeFedCluster(singleAwait.Ns, newName, current.Spec.CABundle,
+			current.Spec.APIEndpoint, current.Spec.SecretRef.Name, labels)
+
+		// when
+		err := awaitility.Client.Create(context.TODO(), newFedCluster, CleanupOptions(ctx))
+
+		// then the KubeFedCluster should not be ready
```

Comment:
It **should** be ready, right? You basically created a new fed cluster resource which points to the same expected-to-be-already-available cluster (current.Spec.APIEndpoint), right? So, the comment is not accurate?

Reply:
yes, you are right. I was struggling with the test configuration for some time and I have changed the tests as well as their order several times, but I forgot to change the comments :-). 
Thanks for pointing this out - fixed: 13307bf